internal/featuretoggle/monitoring_stack_feature_toggle.go

```diff
@@ -0,0 +1,136 @@
+package featuretoggle
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/mt-sre/devkube/dev"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+	addoncontroller "github.com/openshift/addon-operator/internal/controllers/addon"
+)
+
+var _ FeatureToggleHandler = (*MonitoringStackFeatureToggle)(nil)
+
+type MonitoringStackFeatureToggle struct {
+	Client                      client.Client
+	FeatureTogglesInCluster     addonsv1alpha1.AddonOperatorFeatureToggles
+	SchemeToUpdate              *runtime.Scheme
+	AddonReconcilerOptsToUpdate *[]addoncontroller.AddonReconcilerOptions
+}
+
+func (m MonitoringStackFeatureToggle) Name() string {
+	return "Monitoring Stack Reconciliation Feature Toggle"
+}
+
+func (m MonitoringStackFeatureToggle) IsEnabled() bool {
+	return m.FeatureTogglesInCluster.ExperimentalFeatures
```

Comment:
> every feature (which needs to be behind a toggle) would have an attribute under FeatureTogglesInCluster 

Nope that's not the case actually. Until planned otherwise, the existing plan is to serve other feature-toggle functionalities, if they arise, in the same feature toggle of `ExperimentalFeatures` just so that we don't end up in a feature-toggle hell (pretty much an anti-pattern). 
And just like that, even `MonitoringStack` feature is that kind of feature which want to serve as an "experimental feature" and hence, something behind the `ExperimentalFeatures` toggle.